core/src/scene/styleMixer.cpp

```diff
@@ -201,9 +201,6 @@ void StyleMixer::mergeFieldTakingLast(std::string key, Node target, const std::v
 void StyleMixer::mergeMapFieldTakingLast(std::string key, Node target, const std::vector<Node>& sources) {
```

Comment:
Fair enough, refs it is.

On Fri, Mar 11, 2016 at 4:25 PM, Karim Naaji <notifications@github.com>
wrote:

> In core/src/scene/styleMixer.cpp
> <https://github.com/tangrams/tangram-es/pull/619#discussion_r55890763>:
>
> > @@ -201,9 +201,6 @@ void StyleMixer::mergeFieldTakingLast(std::string key, Node target, const std::v
> >  void StyleMixer::mergeMapFieldTakingLast(std::string key, Node target, const std::vector<Node>& sources) {
>
> I'm not convinced we should generally rely that much on that,
> quick bench shows it's much faster to pass by const ref, even with
> optimization on:
> https://gist.github.com/karimnaaji/d58afb771de5edfc98a1
>
> ~/Desktop/cpp-benchs  master ✗                                                                                                                                               80d ⚑
> > g++ const_string_vs_value_string.cpp -O3 -std=c++11
>
> ~/Desktop/cpp-benchs  master ✗                                                                                                                                               80d ⚑
> > ./a.out
> function_a: 0.002ms         # const ref
> function_b: 68.776ms       # copy
> function_b-move: 2.86ms # move
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tangrams/tangram-es/pull/619/files#r55890763>.
>